website/addons/base/__init__.py

```diff
@@ -602,6 +602,8 @@ def get_attached_nodes(self, external_account):
         for node in self.get_nodes_with_oauth_grants(external_account):
             if node is None:
                 continue
+            if node.is_deleted:
```

Comment:
It might make sense to filter out deleted nodes in `get_nodes_with_oauth_grants` instead of here. Are there any places we call that method where we would want to include deleted nodes?